### ISSUE TITLE
Yup array validation is not validating array correctly

### ISSUE THREAD

**Issue by geordyjames** [jquense/yup#603]
Yup array validation is not validating array correctly. But both **Array.isArray** and **lodash isArray** is validating array correctly. Since yup is directly using lodash as its dependency, I hope you will release a bug fix soon.

Here is the code sample for a better explanation.

```
const yup = require('yup');
const isArray = require('lodash/isArray');

const userIds = '[1, 2]'; // or "[1, 2]" is accepting as valid array in yup.array()

console.log('Result from Array.isArray ' + Array.isArray(userIds));

console.log('Result from lodash isArray ' + isArray(userIds));

const schema = yup.object({
	userIds: yup.array()
});

schema
	.validate({
		userIds
	})
	.then(result => console.log(result))
	.catch(err => console.log(err));

```

As you can see that array enclosed in **single quotes or double quotes** are not properly validating.

**Comment by jayarjo** [comment]
I just stumbled upon array mystery as well. And I blame it on this line: https://github.com/jquense/yup/blob/master/src/array.js#L25. Why on earth would `yup` attempt to `JSON.parse` it?

**Comment by jayarjo** [comment]
Ok, from what I see it has been in there from very first release: https://github.com/jquense/yup/blob/9012e806f2852285b32edddd855606ab57014cd6/src/array.js#L18. So probably not accidental, but rather a part of deep internal philosophy. Wouldn't mind to hear the reasons behind it though. 'Cause it's not exactly intuitive and caused an anomaly in the logic flow today.

**Comment by geordyjames** [comment]
@jquense This issue is still not fixed in yup version 0.28.1. I checked and verify it myself.
